pkg/validation/spec/swagger_test.go

```diff
@@ -126,3 +141,94 @@ func TestSwaggerSpec_Deserialize(t *testing.T) {
 		assert.EqualValues(t, actual, spec)
 	}
 }
+
+func TestSwaggerSpec_ExperimentalUnmarshal(t *testing.T) {
+	fuzzer := fuzz.
+		NewWithSeed(1646791953).
+		NilChance(0.8).
+		MaxDepth(10).
+		NumElements(1, 2)
+
+	fuzzer.Funcs(
+		SwaggerFuzzFuncs...,
+	)
+
+	expected := Swagger{}
+	fuzzer.Fuzz(&expected)
+
+	// Serialize into JSON
+	jsonBytes, err := json.Marshal(expected)
+	require.NoError(t, err)
+
+	t.Log("Specimen", string(jsonBytes))
+
+	actual := Swagger{}
+	internal.UseExperimentalJSONUnmarshaling = true
+	err = json.Unmarshal(jsonBytes, &actual)
+	require.NoError(t, err)
+
+	control := Swagger{}
+	internal.UseExperimentalJSONUnmarshaling = false
+	err = json.Unmarshal(jsonBytes, &control)
```

Comment:
You're not doing anything with control?

Reply:
I did have a comparison but had found a bug in the jsonv1 unmarshalling that was not present in jsonv2 unmarshal.

I've commented it out for now

Reply:
I'm curious, what was the bug?

Reply:
Fixed by https://github.com/kubernetes/kube-openapi/pull/320

But not with library, but with our usage. If there are any vendor extensions among the fields of a `Responses` object, then the decoding with our old implementation would fail. This is because it tried to decode entire object into `map[string]Response`. I've added logic to only decode interesting fields.

Reply:
control now used in comparison since 320 merged